Makefile

```diff
@@ -12,6 +12,10 @@ OUT_DIR=$(shell pwd)/build/.out
 
 .DEFAULT_GOAL := help
 
+AGENT_VERSION ?= 2.22.1
```

Comment:
Is there a reason to pin a specific Agent version as opposed to pulling latest at build time or set a minimum?

Reply:
this will prevent any potential side effects of a silent upgrade of a version - like breaking changes

Reply:
I get it.  But that should not be possible given the system.
At the same time, it seems like breaking changes without compatibility would not be in the best interest of the Agent project.  I would rather push the requirement into them of not breaking but supporting compatibility for a period of time.

The concern comes in not maintaining dependencies and updates.

Reply:
I prefer the determinism of specifying the version, especially while we are working on integrating the agent. I can see changing this to be the latest release once we've bumped the agent version a few times without issues.